piDIOComm.c

```diff
@@ -116,38 +116,32 @@ INT32U piDIOComm_Config(uint8_t i8uAddress, uint16_t i16uNumEntries, SEntryInfo
 
 INT32U piDIOComm_Init(INT8U i8uDevice_p)
 {
-	u8 addr = RevPiDevice_getDev(i8uDevice_p)->i8uAddress;
 	int ret;
-	SIOGeneric sResponse_l;
-	INT8U i, len_l;
+	INT8U i;
+	u8 *snd_buf;
 
 	pr_info_dio("piDIOComm_Init %d of %d  addr %d numCnt %d\n", i8uDevice_p, i8uConfigured_s,
 		    RevPiDevice_getDev(i8uDevice_p)->i8uAddress,
 		    i8uNumCounter[RevPiDevice_getDev(i8uDevice_p)->i8uAddress]);
 
+	ret = 4;  // unknown device
+
 	for (i = 0; i < i8uConfigured_s; i++) {
-		if (dioConfig_s[i].uHeader.sHeaderTyp1.bitAddress == RevPiDevice_getDev(i8uDevice_p)->i8uAddress) {
-			ret = piIoComm_send((INT8U *) & dioConfig_s[i], sizeof(SDioConfig));
-			if (ret == 0) {
-				len_l = 0;	// empty config telegram
-
-				ret = piIoComm_recv((INT8U *) & sResponse_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-				if (ret > 0) {
-					if (piIoComm_response_valid(&sResponse_l, addr, len_l)) {
-						return 0;	// success
-					} else {
-						return 1;	// wrong crc
-					}
-				} else {
-					return 2;	// no response
-				}
-			} else {
-				return 3;	// could not send
-			}
+		if (dioConfig_s[i].uHeader.sHeaderTyp1.bitAddress ==
+		    RevPiDevice_getDev(i8uDevice_p)->i8uAddress) {
+			snd_buf = ((u8 *) &dioConfig_s[i]) + IOPROTOCOL_HEADER_LENGTH;
```

Comment:
Scope `snd_buf` to the if-clause, i.e. `u8 *snd_buf = ...` to avoid the `u8 *snd_buf;` declaration at the top of the function.

Add `u8 snd_len = sizeof(dioConfig_s[i]) - IOPROTOCOL_HEADER_LENGTH - 1;` here and use it in the call to `pibridge_req_io()` below, thus not exceeding 80 chars per line.

Reply:
I prefer to have all variables at the beginning of the function. However I rearranged the code to not exceed the 80 chars limit.